commit 6f35d2835268eade059535b62378d6d407ef9e87
Author: Sage Weil <sweil@redhat.com>
Date:   Fri Sep 27 16:01:44 2019 -0500

    mgr,mgr/MgrClient: use fsid to signal mon-mgr vs cli MCommands
    
    We can't use the feature bit for the MCommand connection to tell whether
    it is a tell or CLI command because new clients may have to send CLI
    commands via MCommand for old clusters, and they don't always know whether
    this mgr is new or old yet.
    
    Prior to octopus, MCommand contained a mon/mgr CLI command, and did not
    have the fsid field set.  Start populating the fsid field, and use this
    to signal whether a client is a new MgrClient that knows MCommand vs
    MMgrCommand.  If we get an MCommand with the fsid set, that means it is
    a tell command; otherwise, it's an old client sending a CLI command.
    
    Signed-off-by: Sage Weil <sage@redhat.com>

diff --git a/src/mgr/MgrStandby.cc b/src/mgr/MgrStandby.cc
index 465af9299ec..7d193b6575b 100644
--- a/src/mgr/MgrStandby.cc
+++ b/src/mgr/MgrStandby.cc
@@ -48,7 +48,7 @@ MgrStandby::MgrStandby(int argc, const char **argv) :
 		     0)),
   objecter{g_ceph_context, client_messenger.get(), &monc, NULL, 0, 0},
   client{client_messenger.get(), &monc, &objecter},
-  mgrc(g_ceph_context, client_messenger.get()),
+  mgrc(g_ceph_context, client_messenger.get(), &monc.monmap),
   log_client(g_ceph_context, client_messenger.get(), &monc.monmap, LogClient::NO_FLAGS),
   clog(log_client.create_channel(CLOG_CHANNEL_CLUSTER)),
   audit_clog(log_client.create_channel(CLOG_CHANNEL_AUDIT)),
